homeassistant/components/launch_library/diagnostics.py

```diff
@@ -16,14 +19,23 @@ async def async_get_config_entry_diagnostics(
     entry: ConfigEntry,
 ) -> dict[str, Any]:
     """Return diagnostics for a config entry."""
+
     coordinator: DataUpdateCoordinator[LaunchLibraryData] = hass.data[DOMAIN]
     if coordinator.data is None:
         return {}
-    next_launch = coordinator.data["upcoming_launches"][0]
-    starship_launch = coordinator.data["starship_events"].upcoming.launches[0]
-    starship_event = coordinator.data["starship_events"].upcoming.events[0]
+
+    def _first_element(data: list[Launch | Event]) -> dict[str, Any] | None:
+        if not data:
+            return None
+        return data[0].raw_data_contents
+
+    next_launch = _first_element(coordinator.data["upcoming_launches"])
+    starship_launch = _first_element(
+        coordinator.data["starship_events"].upcoming.launches
+    )
+    starship_event = _first_element(coordinator.data["starship_events"].upcoming.events)
```

Comment:
No need to store these here, just call _first_element directly in the return dict